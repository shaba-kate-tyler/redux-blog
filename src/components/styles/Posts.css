.Posts {
    background-color: rgb(244, 244, 244);
    margin: 2vh;
    align-items: center;
    list-style: none;
    
    li:hover {
        background-color: rgba(255, 255, 114, 0.2);
        transition: 0.2s;
        cursor: pointer;
    }
}