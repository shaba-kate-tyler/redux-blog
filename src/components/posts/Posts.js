import React from 'react';
import Post from './Post';
import PropTypes from 'prop-types';
import styles from '../styles/Posts.css';

function Posts({ posts }) {
  const postsList = posts.map(post => 
    <li key={post.id}>
      <Post
        title={post.title}
        body={post.body}
        id={post.id}
      />
    </li>
  );

  return (
    <ul className={styles.Posts}>
      {postsList}
    </ul>
  );
}

Posts.propTypes = {
  posts: PropTypes.array.isRequired
};

export default Posts;
