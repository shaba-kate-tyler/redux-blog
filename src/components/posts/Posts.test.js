import React from 'react';
import renderer from 'react-test-renderer';
import Posts from './Posts';
import { MemoryRouter } from 'react-router-dom';

describe('Posts', () => {
  it('matches a snapshot of Posts', () => {
    const posts = [
      {
        id: 1,
        userId: 10,
        title: 'a post',
        body: 'This is a post'
      },
      {
        id: 2,
        userId: 11,
        title: 'another post',
        body: 'This is a post'
      },
      {
        id: 3,
        userId: 11,
        title: 'another other post',
        body: 'This is a post'
      }
    ];
    const tree = renderer.create(
      <MemoryRouter>
        <Posts
          posts={posts}
        />
      </MemoryRouter>
    );

    expect(tree).toMatchSnapshot();
  });
});
