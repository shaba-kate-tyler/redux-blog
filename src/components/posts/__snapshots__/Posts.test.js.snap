// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts matches a snapshot of Posts 1`] = `
<ul
  className="Posts"
>
  <li>
    <div
      className="Post"
    >
      <a
        href="/posts/1"
        onClick={[Function]}
      >
        <h2>
          a post
        </h2>
      </a>
      <p>
        This is a post
      </p>
    </div>
  </li>
  <li>
    <div
      className="Post"
    >
      <a
        href="/posts/2"
        onClick={[Function]}
      >
        <h2>
          another post
        </h2>
      </a>
      <p>
        This is a post
      </p>
    </div>
  </li>
  <li>
    <div
      className="Post"
    >
      <a
        href="/posts/3"
        onClick={[Function]}
      >
        <h2>
          another other post
        </h2>
      </a>
      <p>
        This is a post
      </p>
    </div>
  </li>
</ul>
`;
